lib/ansible/modules/cloud/docker/docker_network.py

```diff
@@ -62,13 +62,22 @@
     aliases:
       - incremental
 
+  enable_ipv6:
+    version_added: 2.8
+    description:
+      - Enable IPv6 networking.
+    type: bool
+    default: null
+    required: false
+
   ipam_driver:
     description:
       - Specify an IPAM driver.
 
-  ipam_options:
```

Comment:
This is a breaking change to existing functionality. Rather than outright replacing the option here, why not create `ipam_config` as a new option and include logic to use it preferentially if it's set? That way, upgrading users won't have their playbooks break immediately.

Additionally, the `ipam_options` option should be marked as deprecated when doing so.

Reply:
Simply removing an existing option is not acceptable. Adding a new option and deprecating the old one is the best way to do this correctly. (Also, you should include some logic which forbids to use both.)

Reply:
I see your point. This is a bit sad, because the parameters then differ from the official docker specification. So I will add a deprecation note for `ipam_options` and implement the "real" ipam options as parameter `ipam_driver_options`.

Reply:
Why do the parameters differ?

You can still name the new parameter `ipam_options`, as long as you keep the old one (marked as deprecated, to be removed in Ansible 2.12) and make sure that specifying both doesn't lead to problems (i.e. simply fail if both are specified). Internally, you can convert the old one to the new one if the old one is specified.

Reply:
In Docker 1.10.0 (https://docs.docker.com/release-notes/docker-engine/#1100-2016-02-04, https://github.com/moby/moby/pull/17316) the new parameter `--ipam-opt` was introduced. But because the docker_network module already defined the parameter `ipam_options` (which actually is called ipam config in the docker specification) I cannot use `ipam_options` for the newly introduced parameter.

Reply:
Ok, I was mainly confused because a third (not yet mentioned) name showed up. So the idea is to:
 1. deprecate `ipam_options`;
 2. add `ipam_config`, which is auto-created from `ipam_options` if that one is specified;
 3. add `ipam_driver_options` (for the IPAM driver-specific options).
That sounds OK for me.

Otherwise, we could maybe also proceed as follows: deprecate `ipam_driver` and `ipam_options`, and add `ipam` parameter, which accepts something like [here](https://docs.docker.com/compose/compose-file/compose-file-v2/#ipam). This requires a lot more error-checking on the module's side, though, since Ansible won't do much (except guaranteeing that `ipam` is a `dict`).

Reply:
I tried to implement the parameter `ipam_driver_options` (requires docker-py >= 2.0.0), but it requires a lot of changes to the docker_network module to guarantee backward compatibility. So I decided not to implement this feature.